File: src/main/java/upbrella/be/umbrella/dto/response/UmbrellaStatisticsResponse.java
package upbrella.be.umbrella.dto.response;

import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class UmbrellaStatisticsResponse {

    private long totalUmbrellaCount;
    private long rentableUmbrellaCount;
    private long rentedUmbrellaCount;
    private long missingUmbrellaCount;
    private double missingRate;
    private long totalRentCount;

    public static UmbrellaStatisticsResponse fromCounts(long totalUmbrellaCount, long rentableUmbrellaCount,
                                                        long rentedUmbrellaCount, long missingUmbrellaCount,
                                                        long totalRentCount) {

        return UmbrellaStatisticsResponse.builder()
                .totalUmbrellaCount(totalUmbrellaCount)
                .rentableUmbrellaCount(rentableUmbrellaCount)
                .rentedUmbrellaCount(rentedUmbrellaCount)
                .missingUmbrellaCount(missingUmbrellaCount)
                .missingRate((double) 100 * missingUmbrellaCount / totalUmbrellaCount)
                .totalRentCount(totalRentCount)
                .build();
    }
}

File: src/main/kotlin/upbrella/be/umbrella/dto/response/UmbrellaStatisticsResponse.kt
package upbrella.be.umbrella.dto.response

data class UmbrellaStatisticsResponse(
    val totalUmbrellaCount: Long,
    val rentableUmbrellaCount: Long,
    val rentedUmbrellaCount: Long,
    val missingUmbrellaCount: Long,
    val missingRate: Double,
    val totalRentCount: Long
) {
    companion object {
        fun fromCounts(
            totalUmbrellaCount: Long,
            rentableUmbrellaCount: Long,
            rentedUmbrellaCount: Long,
            missingUmbrellaCount: Long,
            totalRentCount: Long
        ): UmbrellaStatisticsResponse {
            val missingRate = if (totalUmbrellaCount != 0L) {
                100.0 * missingUmbrellaCount / totalUmbrellaCount
            } else {
                0.0
            }

            return UmbrellaStatisticsResponse(
                totalUmbrellaCount = totalUmbrellaCount,
                rentableUmbrellaCount = rentableUmbrellaCount,
                rentedUmbrellaCount = rentedUmbrellaCount,
                missingUmbrellaCount = missingUmbrellaCount,
                missingRate = missingRate,
                totalRentCount = totalRentCount
            )
        }
    }
}